test/src/test/java/org/corfudb/integration/LogSizeQuotaIT.java

```diff
@@ -0,0 +1,102 @@
+package org.corfudb.integration;
+
+import org.corfudb.infrastructure.log.StreamLogFiles;
+import org.corfudb.protocols.wireprotocol.Token;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.MultiCheckpointWriter;
+import org.corfudb.runtime.collections.CorfuTable;
+import org.corfudb.runtime.exceptions.QuotaExceededException;
+import org.corfudb.runtime.object.transactions.Transaction;
+import org.corfudb.runtime.view.Priority;
+import org.corfudb.runtime.view.stream.IStreamView;
+import org.junit.Test;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletionException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+public class LogSizeQuotaIT extends AbstractIT {
+
+    @Test
+    public void testLogSizeQuota() throws Exception {
+
+        final int numWrites = 1000;
+        final int payloadSize = 100;
+
+        Process server_1 = new CorfuServerRunner()
+                .setHost(DEFAULT_HOST)
+                .setPort(DEFAULT_PORT)
+                .setSingle(true)
+                .setLogPath(getCorfuServerLogPath(DEFAULT_HOST, DEFAULT_PORT))
+                .setLogSizeQuota(Long.toString(numWrites * payloadSize))
+                .runServer();
+
+        byte[] payload = new byte[payloadSize];
+        CorfuRuntime rt = new CorfuRuntime(DEFAULT_ENDPOINT).connect();
+
+        Map<String, String> map = rt.getObjectsView()
+                .build()
+                .setStreamName("s1")
+                .setType(CorfuTable.class)
+                .open();
+
+        // Create a map
+        map.put("k1", "v1");
+        map.put("k2", "v2");
+
+        // Fill the log with data till the quota is reached
+        IStreamView sv = rt.getStreamsView().get(UUID.randomUUID());
+
+        boolean quotaReached = false;
+        for (int x = 0; x < numWrites; x++) {
+            try {
+                sv.append(payload);
+            } catch (QuotaExceededException e) {
+                quotaReached = true;
+                break;
+            }
+        }
+
+        assertThat(quotaReached).isTrue();
+
+        // Start/Commit a high priority transaction
+        rt.getObjectsView().TXBuild().priority(Priority.HIGH).build().begin();
+        map.put("k3", "k3");
```

Comment:
If an object sync encounters a hole, the hole filling mechanism won;t tag its write with high priority.